components/sup/src/manager/service/hooks.rs

```diff
@@ -17,116 +17,358 @@ use std::fmt;
 use std::io::BufReader;
 use std::io::prelude::*;
 use std::path::{Path, PathBuf};
-use std::process::Child;
+use std::process::{Child, ExitStatus};
+use std::result;
 
 use hcore;
 use hcore::service::ServiceGroup;
+use serde::{Serialize, Serializer};
 
-use error::{Error, Result};
+use super::health;
+use error::Result;
 use manager::service::ServiceConfig;
 use supervisor::RuntimeConfig;
 use templating::Template;
 use util;
 
 pub const HOOK_PERMISSIONS: u32 = 0o755;
-pub const INIT_FILENAME: &'static str = "init";
-pub const HEALTHCHECK_FILENAME: &'static str = "health_check";
-pub const FILEUPDATED_FILENAME: &'static str = "file_updated";
-pub const RECONFIGURE_FILENAME: &'static str = "reconfigure";
-pub const SMOKETEST_FILENAME: &'static str = "smoke_test";
-pub const RUN_FILENAME: &'static str = "run";
-
 static LOGKEY: &'static str = "HK";
 
-#[derive(Debug, Serialize)]
-pub struct Hook {
-    pub htype: HookType,
-    pub template: PathBuf,
-    pub path: PathBuf,
-    pub user: String,
-    pub group: String,
-}
-
-impl Hook {
-    pub fn new(htype: HookType,
-               template: PathBuf,
-               path: PathBuf,
-               user: String,
-               group: String)
-               -> Self {
-        Hook {
-            htype: htype,
-            template: template,
-            path: path,
-            user: user,
-            group: group,
-        }
+#[derive(Debug, Copy, Clone)]
+pub struct ExitCode(i32);
+
+impl Default for ExitCode {
+    fn default() -> ExitCode {
+        ExitCode(-1)
     }
+}
 
-    /// Run a compiled hook.
-    pub fn run(&self, service_group: &ServiceGroup) -> Result<()> {
```

Comment:
Do we really need to pass `ServiceGroup` around in all the functions. It appears that its only being used for output purposes. Can't we just pass it once on creation?